Build from source command doesn't work

### What did you do?

Followed the README to install Prometheus from source using: `GO111MODULE=on go install github.com/prometheus/prometheus/cmd/prometheus`

### What did you expect to see?

Prometheus being installed.

### What did you see instead? Under which circumstances?

The following compilation error:

```sh
$ GO111MODULE=on go install github.com/prometheus/prometheus/cmd/prometheus/...
/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20220308163432-03831554a519/web/ui/doc.go:19:2: missing go.sum entry for module providing package github.com/shurcooL/httpfs/filter (imported by github.com/prometheus/prometheus/web/ui); to add:
        go get github.com/prometheus/prometheus/web/ui@v1.8.2-0.20220308163432-03831554a519
/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20220308163432-03831554a519/web/ui/doc.go:20:2: missing go.sum entry for module providing package github.com/shurcooL/httpfs/union (imported by github.com/prometheus/prometheus/web/ui); to add:
        go get github.com/prometheus/prometheus/web/ui@v1.8.2-0.20220308163432-03831554a519
/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20220308163432-03831554a519/web/ui/doc.go:21:2: missing go.sum entry for module providing package github.com/shurcooL/vfsgen (imported by github.com/prometheus/prometheus/web/ui); to add:
        go get github.com/prometheus/prometheus/web/ui@v1.8.2-0.20220308163432-03831554a519
```

### System information

Linux codespaces-787b6c 5.4.0-1074-azure #77~18.04.1-Ubuntu SMP Wed Mar 30 15:36:02 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Prometheus version

_No response_

### Prometheus configuration file

_No response_

### Alertmanager version

_No response_

### Alertmanager configuration file

_No response_

### Logs

_No response_

--- a/README.md
+++ b/README.md
@@ -70,7 +70,7 @@ git clone https://github.com/prometheus/prometheus.git
 cd prometheus
 ```
 
-You can directly use the `go` tool to download and install the `prometheus`
+You can use the `go` tool to build and install the `prometheus`
 and `promtool` binaries into your `GOPATH`:
 
 ```bash

--- a/README.md
+++ b/README.md
@@ -66,8 +66,6 @@ To build Prometheus from source code, You need:
 Start by cloning the repository:
 
 ```bash
-mkdir -p $GOPATH/src/github.com/prometheus
-cd $GOPATH/src/github.com/prometheus
 git clone https://github.com/prometheus/prometheus.git
 cd prometheus
 ```

--- a/README.md
+++ b/README.md
@@ -63,6 +63,15 @@ To build Prometheus from source code, You need:
 * NodeJS [version 16 or greater](https://nodejs.org/).
 * npm [version 7 or greater](https://www.npmjs.com/).
 
+Start by cloning the repository:
+
+```bash
+mkdir -p $GOPATH/src/github.com/prometheus
+cd $GOPATH/src/github.com/prometheus
+git clone https://github.com/prometheus/prometheus.git
+cd prometheus
+```
+
 You can directly use the `go` tool to download and install the `prometheus`
 and `promtool` binaries into your `GOPATH`:
 
@@ -79,14 +88,10 @@ React UI unless it has been built explicitly using `make assets` or `make build`
 
 An example of the above configuration file can be found [here.](https://github.com/prometheus/prometheus/blob/main/documentation/examples/prometheus.yml)
 
-You can also clone the repository yourself and build using `make build`, which will compile in
-the web assets so that Prometheus can be run from anywhere:
+You can also build using `make build`, which will compile in the web assets so that
+Prometheus can be run from anywhere:
 
 ```bash
-mkdir -p $GOPATH/src/github.com/prometheus
-cd $GOPATH/src/github.com/prometheus
-git clone https://github.com/prometheus/prometheus.git
-cd prometheus
 make build
 ./prometheus --config.file=your_config.yml
 ```

